geopy/geocoders/arcgis.py

```diff
@@ -83,6 +84,12 @@ def __init__(
             See :attr:`geopy.geocoders.options.default_ssl_context`.
 
             .. versionadded:: 1.14.0
+
+        :param str domain: Domain where the target ArcGIS service
+            is hosted.
+            
+        :param str auth_domain: Domain where the target ArcGIS auth service
```

Comment:
Also I would suggest to add to the auth_domain doc a clarification that this parameter will be used only when authentication is enabled (i.e. username, password or referer are set), otherwise it might be confusing what to put here if a local geocoder server does not have a corresponding auth server.